src/main/java/org/opentripplanner/routing/algorithm/strategies/SimpleEuclideanRWH.java

```diff
@@ -0,0 +1,103 @@
+package org.opentripplanner.routing.algorithm.strategies;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.core.vehicle_sharing.*;
+import org.opentripplanner.routing.edgetype.FreeEdge;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.graph.Vertex;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static java.lang.Double.min;
+import static org.opentripplanner.common.geometry.SphericalDistanceLibrary.fastDistance;
+
+public class SimpleEuclideanRWH implements RemainingWeightHeuristic {
+
+    private double lat;
+    private double lon;
+    private double bestMultiplier;
+
+    @Override
+    public void initialize(RoutingRequest options, long abortTime) {
+        calculateDestination(options.rctx.target);
+        calculateBestMultiplier(options);
+    }
+
+    private void calculateDestination(Vertex target) {
+        if (target.getDegreeIn() == 1) {
```

Comment:
nie rozumiem tego warunku, dlaczego jak target != 0 to zmieniamy lat,lon. A jak taget == 1 i edge == null. to już nie?

Reply:
I don't exactly get that code either, but it is just moved from previous class, and it works, so I didn't get into it.